src/fill_pg_plugin.cpp

```diff
@@ -602,28 +679,50 @@ struct fpg_session : connection_callbacks, std::enable_shared_from_this<fpg_sess
     ts->writer.write_raw_line(boost::algorithm::join(values, "\t"));
   }
 
-  void write_stream_transactions(uint32_t block_num, const std::string& name, const std::vector<std::string>& values) {
-    if (!first_bulk)
-      first_bulk = block_num;
-    auto& ts = table_streams[name];
-    if (!ts) {
-      std::vector<std::string> columns;
-      if (name == "transactions")
-      {
-        columns = {"transaction_number", "block_number", "transaction_ordinal", "id", "status"};
-      }
-      else if (name == "actions")
-      {
-        columns = {"action_number", "transaction_id", "action_ordinal", "creator_action_ordinal", "receiver", "action_account", "action_name", "action_permission", "action_data", "context_free", "console"};
-      }
-      else if (name == "action_data")
-      {
-        columns = {"action_data_number", "action_number", "key", "value"};
-      }
+  void write_stream_custom(uint32_t block_num, const std::string& name, const std::vector<std::string>& values) {
+    try {
+      if (!first_bulk)
+        first_bulk = block_num;
+      auto& ts = table_streams[name];
+      if (!ts) {
+        std::vector<std::string> columns;
+        if (name == "transactions")
+        {
+          columns = {"transaction_number", "block_number", "transaction_ordinal", "id", "status"};
+        }
+        else if (name == "actions")
+        {
+          columns = {"action_number", "transaction_number", "action_ordinal", "creator_action_ordinal", "receiver", "action_account", "action_name", "context_free", "console"};
```

Comment:
maybe initialise these column names and also on line 691 in a separate file, use it as a schema

Reply:
@shannendorothee This is a good suggestion. As it stands, there are tons of refactors we could implement like this, mostly due to the fact that the project this was forked from is open source.

It would be great to improve the whole codebase as it stands, but we don't plan to alter this project once we're done with it for quite some time, so I'm not sure it's worth taking the time to extract code like this to their own separate files. Especially since we need to focus our efforts on the advanced exchange right now.

Reply:
updated